interop-testing/src/test/java/io/grpc/testing/integration/ConcurrencyTest.java

```diff
@@ -117,14 +119,14 @@ public void run() {
         // Wait until all client worker threads are poised & ready, then send the request. This way
         // all clients send their requests at approximately the same time.
         startBarrier.await();
-        clientStub.streamingOutputCall(request, new SignalingResponseObserver(responsesDoneSignal));
-      } catch (Exception e) {
-        throw e instanceof RuntimeException ? (RuntimeException) e : new RuntimeException(e);
+        clientStub.streamingOutputCall(request, new SignalingResponseObserver(completionFuture));
+      } catch (Throwable t) {
+        completionFuture.setException(t);
```

Comment:
Do we need to add handle `InterruptedException` differently to propagate the interrupt?

Reply:
It isn't all that important since this Runnable is passed directly to the executor, and the executor will clear the interruption flag. But yeah, it is good practice.